Code/Framework/AzCore/AzCore/Asset/AssetDataStream.h

```diff
@@ -9,17 +9,26 @@
 
 #include <AzCore/IO/GenericStreams.h>
 #include <AzCore/IO/IStreamerTypes.h>
-#include <AzCore/IO/IStreamer.h>
-#include <AzCore/std/containers/vector.h>
-#include <AzCore/std/parallel/condition_variable.h>
-#include <AzCore/Asset/AssetCommon.h>
-#include <AzCore/Debug/Profiler.h>
+#include <AzCore/std/function/function_template.h>
```

Comment:
`function_template.h` isn't meant to be included outside of `functional.h`.
I believe you want to include `function_fwd.h` here if using an AZStd::function

Reply:
Sadly function_fwd will not work here, since `OnCompleteCallback loadCallback` is a defaulted argument